src/dodal/devices/motors.py

```diff
@@ -59,3 +61,24 @@ def __init__(
             self.phi = Motor(prefix + infix[4])
             self.omega = Motor(prefix + infix[5])
         super().__init__(name=name, prefix=prefix, infix=infix[0:3])
+
+
+class MotorGroup(StandardReadable):
+    """
+    Generic group of motors, dynamically created from a name -> PV dictionary
+    """
+
+    def __init__(self, motor_name_to_pv: dict[str, str], name=""):
+        with self.add_children_as_readables():
+            for raw_motor_name, pv in motor_name_to_pv.items():
+                motor_name = safe_identifier(raw_motor_name)
+                setattr(self, motor_name, Motor(pv, motor_name))
```

Comment:
```suggestion
        with self.add_children_as_readables():
            self.x = Motor(f"{prefix}X")
            self.y = Motor(f"{prefix}T")
```

Then in your beamline module you just need:


```python
@device_factory()
def kb() -> MotorGroup:
    return MotorGroup(f"{PREFIX.beamline_prefix}-OP-KBM-01:CS:")
```

Reply:
Reverted the scientifically-insignificant class